### ISSUE TITLE
Python 3.7

### ISSUE THREAD

**Issue by pbecotte** [noirbizarre/flask-restplus#721]
I see that setup.py was updated to list Python 3.7 the day after the 0.13.0 release- any chance you could roll out an update with that version, even if it is just that commit?  :)

**Comment by Colin-b** [comment]
I would consider it an enhancement given that this is a pure python project (why is there even a wheel ?) and you can always install using the gz ?

**Comment by pbecotte** [comment]
I'm, sure...but presumably this is just a matter of pushing the deploy button? The problem is that the metadata on pypi currently says 3.6 max, so pip install from 3.7 doesn't work. But the setup.py has already been fixed, the change just wasn't released. (Though, 3.8 is immenent, it would probably be helpful to do that while you're at it).

**Comment by SteadBytes** [comment]
@pbecotte It is not just a matter of pushing the deploy button, we would need to include Python 3.8 in our test suite and ensure that Flask-RESTPlus was compatible with it. However I agree, especially since Flask supports up to 3.7 https://github.com/pallets/flask/blob/master/setup.py#L42

**Comment by pbecotte** [comment]
Here, added the extra line to your tox setup, seems to work just fine ;)

https://github.com/noirbizarre/flask-restplus/pull/730

**Comment by SteadBytes** [comment]
@pbecotte we appreciate your enthusiasm, however the Travis build for that PR has failed so it's not working just fine. Also, this issue is about python 3.7, whereas that PR adds python 3.8...

**Comment by pbecotte** [comment]
I mean- someone added 3.7 to the build months ago, so I figured my comment about 3.8 was the objection. I guess I just don't understand why you would update setup.py to handle Python3.7 but not actually release the version after the update was merged. Don't feel like you have to merge my PR or do anything- just trying to be helpful :)

(Also, my PR failed because `3.8` isn't a valid travis tag. Turns out, 3.8-dev is... and is also already part of the build. The tests pass using the 3.8 interpreter).

**Comment by SteadBytes** [comment]
Yeah I'm not 100% sure why a release wasn't made after adding 3.7 support (I've not been able to give much time to the project for a little while however I assume it was because the change was to be included in a larger release with other fixes/improvements.

Apologies if I came off wrong - we definitely appreciate your help :+1: !  However, considering that Flask itself doesn't support Python 3.8 yet I don't think supporting it on PyPi makes much sense :thinking:

**Comment by pbecotte** [comment]
Sure- though the Python3.8 release was scheduled for Monday if nothing has changed. I generally would lean towards letting users install a package with the higher version so long as the tests pass

Could certainly see releasing 3.7 support now, and just holding that PR until Python and Flask do the final update.

**Comment by SteadBytes** [comment]
I agree that letting users install a package with a higher versions makes sense. However, in our case, it *currently* doesn't make sense to install Flask-RESTPlus on Python 3.8 as Flask would couldn't also be installed. That mismatch is confusing and adds a potential non-obvious pitfall to using Flask-RESTPlus. Does that explain my hesitation to release on Python 3.8 *at the moment* ?

As soon as Flask supports Python 3.8 then we should certainly follow suit :+1: 

Yes, Python 3.7 support ought to be released. I'm just waiting for the viewpoint of another maintainer ,@j5awry, as to whether we should make a release *solely* for Python 3.7 support or to include the changes merged into master since Python 3.7 support was added. IMO, the current state of master would be minor version bump from the previous `0.13.0` release to `0.14.0` as there are a number changes that are more than small patch fixes. There may also be other open PRs that could be included in the release.

Please bear with us whilst we coordinate a release, I apologise that the Python 3.7 support wasn't released when it was first merged to master. As I said previously I'm not sure what happened there. However, releasing to PyPi requires a bit of thought/care as to not break any users unintentionally :smile:

**Comment by j5awry** [comment]
I think we can roll it into the next release. Honestly, between this, what's already been merged, and one or two PRs that are ready, we have plenty for a "minor" release. 

Also, we need a real roadmap of what 1.0 looks like...but that's for another ticket.

**Comment by SteadBytes** [comment]
Agreed, any PRs in particular that you would like to get in?

Yes I agree 100% about 1.0 roadmap, that's been out off far too long, but yes, another ticket 🤣
